packages/denylist/src/denylist.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-env serviceworker, browser */
+
+import { normalizeCid } from './utils/cid'
+import { getFromDenyList } from './utils/denylist'
+import { JSONResponse } from '@web3-storage/worker-utils/response'
+
+/**
+ * @param {Array<string>} params
+ * @param {(params: Array<string>, request: Request, env: import('./env').Env) => Promise<Response>} fn
+ * @returns {import('itty-router').RouteHandler<Request>}
+ */
+function withRequiredQueryParams (params, fn) {
```

Comment:
Thoughts on making this API just?

`GET /:cid` ? So, no query parameters. It feels wait more conventional REST to me